### PR TITLE
Add Support for CN0565.

### PR DESCRIPTION
**Pull request by mc-so**
# Description

Adding pyadi-iio support for CN0565 - Electrical Impedance Tomography Measurement System including basic examples.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

The example scripts were tested under Windows OS using EVAL-ADICUP3029 which is an Arduino compatible form factor development board attached to the EVAL-CN0565-ARDZ and its test board.

**Test Configuration**:
* Hardware: EVAL-CN0565-ARDZ + EVAL-ADICUP3029
* OS: Windows 10

# Documentation

Product Page: https://www.analog.com/en/design-center/reference-designs/circuits-from-the-lab/cn0565.html
User Guide: https://wiki.analog.com/resources/eval/user-guides/circuits-from-the-lab/cn0565

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .vscode/launch.json
Comment (tfcollins) [inline .vscode/launch.json:1]: Please remove this file. Right now we don't include editor config files.

File: adi/cn0565.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from adi.context_manager import context_manager
+from adi.ad5940 import ad5940
+from adi.adg2128 import adg2128
+
+class cn0565(ad5940, adg2128, context_manager):
+
+    _device_name = ""
+
+    def __init__(self, uri=""):
+
```
Comment (tfcollins) [inline adi/cn0565.py:11]: Need docstrings on all methods

File: examples/cn0565/.gitignore
Comment (tfcollins) [inline examples/cn0565/.gitignore:1]: Remove as well

File: examples/cn0565/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+To run the GUI:
+python3 main.py -e 16
+
+In case the firmware is running an iio server:
+python3 main.py -b 230400 -e 16 --iio
+
+This will start the GUI for the EIT. The GUI is currently a work in progress so there are a lot of know bugs which I will be fixing later.
+
```
Comment (tfcollins) [inline examples/cn0565/README.md:26]: Please use codeblocks or commandline blocks to separate comments from command or code

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,9 @@
 numpy>=1.20
 pylibiio==0.23.1
 paramiko
+scipy
+matplotlib
+pyserial
+PyQt5
+pyeit
+pandas
```
Comment (tfcollins) [inline requirements.txt:9]: Please move additional packages to the requirements file inside your examples/cn0565 folder

File: test/pyvenv.cfg
Comment (tfcollins) [inline test/pyvenv.cfg:1]: Please remove

File: examples/cn0565/test_script.py
Comment (tfcollins) [inline examples/cn0565/test_script.py:1]: Filename should be more descriptive

File: examples/cn0565/query.py
Comment (tfcollins) [inline examples/cn0565/query.py:1]: Filename should be more descriptive

File: examples/cn0565/eit_dynamic_jac.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# coding: utf-8
+""" demo code for CN0565 EIT - JAC """
+# Copyright (c) Benyuan Liu. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
```
Comment (tfcollins) [inline examples/cn0565/eit_dynamic_jac.py:4]: What is the origin of this file?

File: examples/cn0565/eit_dynamic_greit.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
+# coding: utf-8
+""" demo code for CN0565 EIT - GREIT """
+# Copyright (c) Benyuan Liu. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
```
Comment (tfcollins) [inline examples/cn0565/eit_dynamic_greit.py:4]: What is the origin of this file?

File: examples/cn0565/eit_dynamic_bp.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# coding: utf-8
+""" demo code for CN0565 EIT back-projection """
+# Copyright (c) Benyuan Liu. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
```
Comment (tfcollins) [inline examples/cn0565/eit_dynamic_bp.py:4]: What is the origin of this file?

File: examples/cn0565/cn0565_sample_plot.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
+# coding: utf-8
+""" demo code for CN0565 EIT back-projection """
+# Copyright (c) Benyuan Liu. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
```
Comment (tfcollins) [inline examples/cn0565/cn0565_sample_plot.py:4]: What is the origin of this file?

File: examples/cn0565/cn0565_prod_tst.py [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+    passed = 0
+
+    Electrode_Iter = iter(range(0,16))
+    for neg_e in Electrode_Iter:
+        pos_e = next(Electrode_Iter)   # setting pos_e = Electrode 0 to neg_e
+
+        if pos_e == neg_e: 
+            print("Forcing electrodes are shorted in the given assignments.")
+            return False
```
Comment (tfcollins) [inline examples/cn0565/cn0565_prod_tst.py:73]: I don't really understand what this code does. It just provides odd and even numbers... The conditional can't be reached. Can you explain the idea here?

File: adi/cn0565.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
+import numpy as np
```
Comment (tfcollins) [inline adi/cn0565.py:1]: Needs license+copyright

**Review by tfcollins — CHANGES_REQUESTED** [review]
Be sure to run the linters

File: examples/cn0565/cn0565_prod_tst.py (hunk 8 above)
Comment (mc-so) [reply examples/cn0565/cn0565_prod_tst.py:73]: It is used for labeling, iteration, and positions of the electrode pairs.

Comment (kister-jimenez) [reply examples/cn0565/cn0565_prod_tst.py:78]: Conditional is not reached here. We can remove line 76-78.

File: examples/cn0565/README.md
Comment (kister-jimenez) [inline examples/cn0565/README.md:1]: I can't find the query.py mentioned in this README. Add a short description of the examples. Include also instructions and short description of the examples.

File: .vscode/launch.json
Comment (mc-so) [reply .vscode/launch.json:1]: Done.

File: adi/cn0565.py (hunk 1 above)
Comment (mc-so) [reply adi/cn0565.py]: Added doc strings on all methods.

File: examples/cn0565/.gitignore
Comment (mc-so) [reply examples/cn0565/.gitignore:1]: Removed.

File: examples/cn0565/README.md (hunk 2 above)
Comment (mc-so) [reply examples/cn0565/README.md]: Done adding code blocks.

File: test/pyvenv.cfg
Comment (mc-so) [reply test/pyvenv.cfg:1]: Removed.

File: examples/cn0565/query.py
Comment (mc-so) [reply examples/cn0565/query.py:1]: File name changed to cn0565_eit.py

File: examples/cn0565/eit_dynamic_jac.py (hunk 4 above)
Comment (mc-so) [reply examples/cn0565/eit_dynamic_jac.py]: Updated example and used SPDX.

File: examples/cn0565/eit_dynamic_greit.py (hunk 5 above)
Comment (mc-so) [reply examples/cn0565/eit_dynamic_greit.py]: Updated example and used SPDX. This file was created by us.

File: examples/cn0565/eit_dynamic_bp.py (hunk 6 above)
Comment (mc-so) [reply examples/cn0565/eit_dynamic_bp.py]: Updated example and used SPDX. This file was created by us.

File: examples/cn0565/cn0565_sample_plot.py (hunk 7 above)
Comment (mc-so) [reply examples/cn0565/cn0565_sample_plot.py]: Updated example and used SPDX. This file was created by us.

File: examples/cn0565/cn0565_prod_tst.py (hunk 8 above)
Comment (mc-so) [reply examples/cn0565/cn0565_prod_tst.py]: Removed the if pos_e == neg_e conditional block.

File: adi/cn0565.py (hunk 9 above)
Comment (mc-so) [reply adi/cn0565.py:6]: Added license and copyright

File: requirements.txt (hunk 3 above)
Comment (mc-so) [reply requirements.txt]: Fixed the dependency list.

File: examples/cn0565/test_script.py
Comment (mc-so) [reply examples/cn0565/test_script.py:1]: Renamed to cn0565_prod_tst.py

File: examples/cn0565/README.md
Comment (mc-so) [reply examples/cn0565/README.md:1]: Query was renamed to cn0565_example_single.py

**Comment by kister-jimenez** [comment]
I had a call with Michelle to discuss some changes (i.e rebasing to main, squashing commits, lints, design pattern, etc.) and she fixed it in the latest push.

I'm ok with this.

**Comment by kister-jimenez** [comment]
> I had a call with Michelle to discuss some changes (i.e rebasing to main, squashing commits, lints, design pattern, etc.) and she fixed it in the latest push.
> 
> I'm ok with this.

I just noticed a lot of files. Please rebase commits to the correct main.

**Comment by tfcollins** [comment]
Can you reorder your commits to create a clean rebase?

**Comment by tfcollins** [comment]
Emulation file is missing cn0565.xml. Please run pre-commit checks as well

**Review by tfcollins — APPROVED** [review]

**Pull request merged**